spec/rate_limit_control_spec.rb

```diff
@@ -3,7 +3,49 @@
     expect(RateLimitControl::VERSION).not_to be nil
   end
 
-  it "does something useful" do
-    expect(false).to eq(true)
+  after { Redis.current.flushdb }
+
+  let(:action_config) do
+    {
+      action: 'rate_limit_example',
+      id: 'foobar',
+      allowed_requests: 3,
+      storage: Redis.new,
+      timeout: 10
+    }
+  end
+
+  let(:action) { "Any action you would like to execute" }
+
+  context "when the actions" do
```

Comment:
I think you don't need that `context "when the actions"`.
It's probably better having only:
```ruby
context "when actions exceed the total allowed"
context "when actions don't exceed the total allowed"
context "when there are actions being rate limited"
```

Reply:
I usually do this way when I have two context blocks that start with the same phrase. But in this case I think it's better the way you supposed.